External/FEXCore/Source/Utils/Allocator.cpp

```diff
@@ -131,110 +131,89 @@ namespace FEXCore::Allocator {
     FEX_UNREACHABLE;
   }
 
-  PtrCache* StealMemoryRegion(uintptr_t Begin, uintptr_t End) {
-    PtrCache *Cache{};
-    uint64_t CacheSize{};
-    uint64_t CurrentCacheOffset = 0;
-    constexpr std::array<size_t, 10> ReservedVMARegionSizes = {{
-      // Anything larger than 64GB fails out
-      64ULL * 1024 * 1024 * 1024,   // 64GB
-      32ULL * 1024 * 1024 * 1024,   // 32GB
-      16ULL * 1024 * 1024 * 1024,   // 16GB
-      4ULL * 1024 * 1024 * 1024,    // 4GB
-      1ULL * 1024 * 1024 * 1024, // 1GB
-      512ULL * 1024 * 1024,      // 512MB
-      128ULL * 1024 * 1024,      // 128MB
-      32ULL * 1024 * 1024,       // 32MB
-      1ULL * 1024 * 1024,        // 1MB
-      4096ULL                    // One page
-    }};
-    constexpr size_t AllocationSizeMaxIndex = ReservedVMARegionSizes.size() - 1;
-    uint64_t CurrentSizeIndex = 0;
-
-    int PROT_FLAGS = PROT_READ | PROT_WRITE;
-    for (size_t MemoryOffset = Begin; MemoryOffset < End;) {
-      size_t AllocationSize = ReservedVMARegionSizes[CurrentSizeIndex];
-      size_t MemoryOffsetUpper = MemoryOffset + AllocationSize;
+  static inline uintptr_t flp2 (uintptr_t x)
```

Comment:
C++20 has a `std::bit_floor` function. If that is not available on all our target platforms, would be nice to keep consistent naming (and just define this function `#ifndef __cpp_lib_int_pow2`).

Reply:
aw, that's so cute!

looks like it is missing in 20.04, i'll add an #ifdef'd implementation

Reply:
hmm, __cpp_lib_int_pow2 is defined but but_floor isn't :/